addons/repair/models/repair.py

```diff
@@ -115,8 +115,8 @@ class Repair(models.Model):
         copy=False, help="Sale Order from which the product to be repaired comes from.")
     picking_id = fields.Many2one(
         'stock.picking', 'Return', check_company=True,
-        domain="[('is_repairable', '!=', False), ('company_id', '=', company_id)]",
         copy=False, help="Return Order from which the product to be repaired comes from.")
+    valid_picking_ids = fields.Many2many('stock.picking', compute='_compute_valid_picking_ids')
```

Comment:
The number of picking can be very huge to pass to the web client to make the domain. I would recommend to change it in the view to 
```python
['picking_type_id', 'in', 'allowed_picking_type_id']
```
your computed field can return picking_types instead of picking directly